src/pynxtools/nomad/parser.py

```diff
@@ -142,7 +143,15 @@ def _get_value(hdf_node):
     if hdf_value.dtype.kind in "iufc":
         return hdf_value
     if len(hdf_value.shape) > 0:
-        return str([i for i in hdf_value.astype(str)])
+
+        def recurse(x):
```

Comment:
Why to replace a forward loop with recursion when the shape is known apriori?

Reply:
> Why to replace a forward loop with recursion when the shape is known apriori?

No, we do not know the exact shape.